src/components/utils.ts

```diff
@@ -0,0 +1,65 @@
+import { FlatOption, Option } from "./types";
+
+export const flattenOptions = (array: FlatOption[], options: Option[], path: string[] = []) => {
```

Comment:
I guess another approach would be to have two methods here rather than just the single recursive one, so you could have someting like `getFlattenedOptions` which the client component calls, which then calls the recursive method. That way it could return the flattened options rather than modifying an empty array in place (which is a slightly weird api for initialising), and also the recursive method could take `show` as a parameter (true when called from `getFlattenedOptions`, false when called from itself) - that way you wouldnt' need to calculate isTopLevel for every option.

Reply:
this was a good suggestion thanks!